chart/templates/post-delete-hook-cluster-role.yaml

```diff
@@ -30,11 +30,6 @@ rules:
   - apiGroups: [ "admissionregistration.k8s.io" ]
     resources: [ "validatingwebhookconfigurations", "mutatingwebhookconfigurations" ]
     verbs: [ "get", "list", "delete" ]
-  {{- if .Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy" }}
-  - apiGroups: [ "policy" ]
-    resources: [ "podsecuritypolicies" ]
-    verbs: [ "use", "delete", "create" ]
-  {{- end }}
```

Comment:
Do you mean for Rancher v2.7.2 and K8s v1.24? Yes, we'd need to keep this then.

Reply:
Yep - that's the case that I am worried about. Everything in the post-delete job namespaces would need to not have PSPs for this to work.

Reply:
The same is true for the webhook chart then?

Reply:
The webhook chart only needs to have `use` on the PSPs - it's not trying to remove a PSP. Instead, it just needs to use one so that it can run as root - see the [cluster-role](https://github.com/rancher/webhook/blob/72d2039f1472822b20afa685a6add252c583f2d9/charts/rancher-webhook/templates/pre-delete-hook-cluster-role.yaml#L20-L23) for the relevant lines.

Reply:
Yeah, I also removed that part where a use permission is listed on the cluster role for the webhook, it needs to be brought back.

Reply:
Also, in my tests, I noticed that the cluster role does not need the permission to use the PSP (that block was removed), and the job did not fail. But I think it's because the job was allowed to run as root because the API server did not have the admission controller enabled, effectively making PSPs nonexistent (that's my assumption, so could be wrong). 
So if customers actually have their local cluster setup with the controller enabled, then PSPs will work, and the job will need that permission.

Reply:
> Yeah, I also removed that part where a use permission is listed on the cluster role for the webhook, it needs to be brought back.

I don't agree. The use part of the cluster role allows the pre-delete job to deploy using it's PSP. Since this will be removed, we don't need this anymore (pending discussion of how this works in hardened clusters). This is different than the core rancher/rancher job - when it uninstalls it will create/delete the pre-delete PSP so it needs to have Create, Delete (to create/delete the webhook's pre-delete PSP) and use (to use it's own PSP so it can run).

> Also, in my tests, I noticed that the cluster role does not need the permission to use the PSP (that block was removed), and the job did not fail. But I think it's because the job was allowed to run as root because the API server did not have the admission controller enabled, effectively making PSPs nonexistent (that's my assumption, so could be wrong).
So if customers actually have their local cluster setup with the controller enabled, then PSPs will work, and the job will need that permission.

Right - in a non-hardened cluster I don't think it needs to use the PSP.

Reply:
We may want to still consider removing the "use" part of the rbac. With the PSP removed there's nothing for the post-delete job to use.